modules/mono/csharp_script.cpp

```diff
@@ -2862,8 +2862,10 @@ CSharpScript::CSharpScript() {
 
 CSharpScript::~CSharpScript() {
 #ifdef DEBUG_ENABLED
-	MutexLock lock(CSharpLanguage::get_singleton()->script_instances_mutex);
-	CSharpLanguage::get_singleton()->script_list.remove(&this->script_list);
+	{
+		MutexLock lock(CSharpLanguage::get_singleton()->script_instances_mutex);
+		CSharpLanguage::get_singleton()->script_list.remove(&this->script_list);
```

Comment:
```suggestion
		CSharpLanguage::get_singleton()->script_list.remove(&script_list);
```

Reply:
Should we change that in the constructor, too, while we're at it?

Reply:
I was going to look at removing several invalid `this` uses in this file but go ahead here 🙂